utilities/scan_logs.py

```diff
@@ -0,0 +1,30 @@
+from keywords.utils import log_info
+from keywords.utils import log_error
+
+
+def scan_for_errors(error_strings, log_file_path):
+    """
+    Scans a log file line by line for a provided array of words.
+    We use this to look for errors, so we expect that no words will be found
+    If any of the words are found, we raise an exception.
+
+    'error_strings' should be a list. Example ['panic', 'error']
+    """
+
+    if type(error_strings) != list:
+        raise ValueError("'error_strings must be a list'")
+
+    log_info("Looking for {} in {} ...".format(error_strings, log_file_path))
+
+    # Scan each line in the log file for the words to search for
+    with open(log_file_path) as f:
```

Comment:
This might blow up if the log file is huge, because I believe it will need to load the entire file contents in memory.  Using a [generator approach](http://www.dabeaz.com/generators/Generators.pdf) might be way more efficient here.

Any chance you can test on a 1 GB logfile and make sure that it can process it quickly?  If not, then I think we should consider switching to generators.